src/reducers/console.js

```diff
@@ -73,14 +73,17 @@ export default function console(stateIn, {type, payload, meta}) {
       return initialState;
     case 'CONSOLE_INPUT_CHANGED':
       return state.set('currentInputValue', payload.value);
-    case 'CONSOLE_LOG_PRODUCED':
-      return state.setIn(
-        ['history', meta.key],
-        new ConsoleEntry({
-          value: payload.value,
-          evaluatedByCompiledProjectKey: payload.compiledProjectKey,
-        }),
-      );
+    case 'CONSOLE_LOG_BATCH_PRODUCED': {
+      const newHistory = state.get('history').withMutations((map) => {
+        payload.entries.forEach(({value, compiledProjectKey, key}) => {
+          map.set(key, new ConsoleEntry({
+            value,
+            evaluatedByCompiledProjectKey: compiledProjectKey,
+          }));
+        });
+      });
+      return state.set('history', newHistory);
```

Comment:
This is the smallest thing, but I think in this situation it’s more elegant to use `update()` like so:

```js
return state.update('history', history => history.withMutations((map) => {
  payload.entries.forEach(({value, compiledProjectKey, key}) => {
    map.set(key, new ConsoleEntry({
      value,
      evaluatedByCompiledProjectKey: compiledProjectKey,
    }));
  });
}));
```

Just avoids the need to create the intermediate `newHistory` variable.